engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/hydration/NadelHydrationInputBuilder.kt

```diff
@@ -1,172 +1,159 @@
 package graphql.nadel.enginekt.transform.hydration
 
-import graphql.language.ArrayValue
-import graphql.language.BooleanValue
-import graphql.language.FloatValue
-import graphql.language.IntValue
 import graphql.language.NullValue
-import graphql.language.ObjectField
-import graphql.language.ObjectValue
-import graphql.language.StringValue
-import graphql.language.Value
 import graphql.nadel.enginekt.blueprint.NadelHydrationFieldInstruction
 import graphql.nadel.enginekt.blueprint.hydration.NadelHydrationActorInputDef
 import graphql.nadel.enginekt.blueprint.hydration.NadelHydrationActorInputDef.ValueSource
 import graphql.nadel.enginekt.blueprint.hydration.NadelHydrationStrategy
 import graphql.nadel.enginekt.transform.artificial.NadelAliasHelper
 import graphql.nadel.enginekt.transform.result.json.JsonNode
 import graphql.nadel.enginekt.transform.result.json.JsonNodeExtractor
-import graphql.nadel.enginekt.util.AnyList
-import graphql.nadel.enginekt.util.AnyMap
-import graphql.nadel.enginekt.util.asJsonMap
 import graphql.nadel.enginekt.util.emptyOrSingle
 import graphql.nadel.enginekt.util.flatten
-import graphql.nadel.enginekt.util.mapFrom
+import graphql.nadel.enginekt.util.makeNormalizedInputValue
+import graphql.nadel.enginekt.util.toMapStrictly
+import graphql.nadel.enginekt.util.javaValueToAstValue
 import graphql.normalized.ExecutableNormalizedField
 import graphql.normalized.NormalizedInputValue
-import graphql.schema.GraphQLArgument
-import graphql.schema.GraphQLTypeUtil
 
-internal typealias AnyAstValue = Value<*>
+internal class NadelHydrationInputBuilder private constructor(
+    private val instruction: NadelHydrationFieldInstruction,
+    private val aliasHelper: NadelAliasHelper,
+    private val fieldToHydrate: ExecutableNormalizedField,
+    private val parentNode: JsonNode,
+) {
+    companion object {
+        fun getInputValues(
```

Comment:
I think we should follow this pattern more.

One "static" function to take in all the arguments. Create a class that holds some global ish variables that would otherwise need to be passed around to all functions.

This REALLY clears up what a function really needs and what you need to pass around e.g.

```kotlin
    private fun makeInputValue(
        inputDef: NadelHydrationActorInputDef,
    ): NormalizedInputValue? {
```

Is much, much better than

```kotlin
    private fun makeInputValue(
        actorInputDef: NadelHydrationActorInputDef,
        argumentDef: GraphQLArgument,
        parentNode: JsonNode,
        hydrationField: ExecutableNormalizedField,
        aliasHelper: NadelAliasHelper,
    ): NormalizedInputValue? {
```